packages/@aws-cdk/cdk/lib/core/tag-manager.ts

```diff
@@ -0,0 +1,157 @@
+import { Construct } from './construct';
+import { Token } from './tokens';
+
+/**
+ * ITaggable indicates a entity manages tags via the `tags` property
+ */
+export interface ITaggable {
+    tags: TagManager,
+}
+
+/**
+ * A single tag property values
+ */
+interface ITagProps {
+    value: string | Token;
+    propagate?: boolean;
+}
+
+/**
+ * A an object of tags
+ */
+export interface ITags {
+    [key: string]: ITagProps;
+}
+
+/**
+ * A single tag
+ */
+export interface ITag {
+    key: string;
+    value: string | Token;
+    propagate?: boolean;
+}
+
+/**
+ * TagManager facilitates a common implementation of tagging for Constructs.
+ *
+ * Each construct that wants to support tags should implement the `ITaggable`
+ * interface and properly pass tags to the `Resources` (Cloudformation) elements
+ * the `Construct` creates.
+ */
+export class TagManager {
+
+    /**
+     * Checks if the object implements the `ITaggable` interface
+     */
+    public static isTaggable(taggable: ITaggable | any): taggable is ITaggable {
+        return ((taggable as ITaggable).tags !== undefined);
+    }
+
+    private readonly tags: ITags = {};
+
+    constructor(readonly parent: Construct, initialTags: ITags = {}) {
+        this.tags = initialTags;
+
+    }
+
+    /**
+     * Converts the `tags` to a Token for use in lazy evaluation
+     */
+    public toToken(propgateOnly = false): Token {
+        return new Token(() => this.toArray(propgateOnly));
+    }
+
+    /**
+     * Returns a deep copy of the tags as `Array<{key: string, value: any}>`
+     */
+    public toArray(propgateOnly = false): ITag[] {
+        let keys = Object.keys(this.tags);
+        if (propgateOnly) {
+            keys = keys.filter( (key) => {
+                return this.tags[key].propagate === true;
+            });
+        }
+        return keys.map( key => ({key, value: this.tags[key].value}));
+    }
+
+    /**
+     * Merges tags and returns a new ITags object.
+     *
+     * If tag collision occurs the passed in tags are taken. The propagate value
+     * is defaulted to true if it is not provided. This is useful to tag tags
```

Comment:
change `This is useful to tag tags ...` to `This is useful to combine tags from a parent with user provided tag list before creating a new resource`

Reply:
Obsolete